source/MapPanel.cpp

```diff
@@ -1386,10 +1398,36 @@ void MapPanel::DrawSystems()
 
 	// Draw the circles for the systems.
 	double zoom = Zoom();
+
+	const float ringFade = isStarry ? 1.5 - 1.25 * zoom : 1.;
 	for(const Node &node : nodes)
 	{
 		Point pos = zoom * (node.position + center);
-		RingShader::Draw(pos, OUTER, INNER, node.color);
+
+		// System rings fade as you zoom in if starry map is enabled.
+		RingShader::Draw(pos, OUTER, INNER, node.color.Additive(max(ringFade,
+			node.mapIcon.empty() ? .9f : 0)));
+
+		if(isStarry)
+		{
+			// Ensures every multiple-star system has a characteristic, deterministic rotation.
+			float starAngle = node.name.length() + node.position.Length();
+			float spin = 4 * acos(0.) / node.mapIcon.size();
+			Point starOffset = (node.mapIcon.size() == 1) ? Point(0, 0) : node.mapIcon.size() * Point(2, 2);
+
+			// Draw the star sprites
+			for(const Sprite *star : node.mapIcon)
+			{
+				starAngle = starAngle + spin;
+				Point starRotate(cos(starAngle), sin(starAngle));
```

Comment:
I strongly suggest you use the existing `Angle` class to represent angles, instead of storing them in floats and reinventing things like getting a unit `Point` for an angle (which is trivial when using the `Angle` class).